test/test_network.h

```diff
@@ -326,12 +326,12 @@ TEST(network, set_netphase) {
 TEST(network, test) {
   network<sequential> net;
   fully_connected_layer fc(30, 1);
-  int data_num = 300;
+  fc.weight_init(parameter_init::constant(1.0));
+  fc.init_parameters();
```

Comment:
Why did you remove this from previous tests but keep it here?

Reply:
Layer API has this behaviour since always. `weight_init()` and `bias_init()` set the initialization strategies but not perform initialization. So we need to call `layer.setup(true)` or `layer.init_parameters()` to initialize them.

But for network, we need not require to explicitly call `network.init_parameters()` as initialization happens in layers through `<<` operator. Although we explicitly call `init_parameters` when we set different initialization strategies. (Default ones are `xavier` for weights and `constant(0)` for biases.)

Reply:
Previous `layer.setup(true)` was called but calling that during `<<` operator network makes sense as there is data allocation to incoming and outgoing edges of layer. But If I initialize a single layer object, I thought calling `init_parameters()` instead of `setup()` shows more clarity in what we actually intended to do.